src/Store.java

```diff
@@ -0,0 +1,168 @@
+        Bill[] BillsCopy = new Bill[Bills.length];
+        for (int i = 0; i < Bills.length; i++) {
+            BillsCopy[i] = new Bill(Bills[i].Nominal, Bills[i].Amount);
+        }
+
+        int change = amount;
+        boolean stopper = true;
+        while (stopper) {
+            int copyOfChange = change;
+            for (int i = 5; i >= 0; i--) {
+                Bill theBill = BillsCopy[i];
+                if (change >= theBill.Nominal && theBill.Amount != 0) {
+                    change = change - theBill.Nominal;
+                    theBill.Amount -= 1;
+                }
+            }
+            if (change == 0) {
+                String localReport = makeGiveReport(Bills, BillsCopy);
+                System.out.println(localReport);
+                Bills = BillsCopy;
+                stopper = false;
+                addToLog(" Given " + amount + " UAH to a user");
+            } else if (copyOfChange == change) {
+                stopper = false;
+                System.out.println("There are no enough Bills !");
+            }
+        }
+    }
+
+    /**
+     * @param amount - amount of money which user wish to withdraw
+     */
+    public static void giveMoney(int amount) {
+        Bill[] BillsCopy = new Bill[Bills.length];
+        for (int i = 0; i < Bills.length; i++) {
+            BillsCopy[i] = new Bill(Bills[i].Nominal, Bills[i].Amount);
+        }
+        int change = amount;
+        for (int i = 5; i >= 0; i--) {
+            Bill theBill = BillsCopy[i];
+            int amountOfTheBills = change / theBill.Nominal;
+            if (change >= theBill.Nominal && theBill.Amount != 0) {
+                if (amountOfTheBills > theBill.Amount) {
+                    change = change - theBill.Nominal * theBill.Amount;
+                    theBill.Amount = 0;
+                } else {
+                    change = change - theBill.Nominal * amountOfTheBills;
+                    theBill.Amount -= amountOfTheBills;
+                }
+            } else if (theBill.Amount != 0) {
+                change = change - theBill.Nominal * amountOfTheBills;
+                theBill.Amount -= amountOfTheBills;
+            }
+        }
+        if (change == 0) {
+            String localReport = makeGiveReport(Bills, BillsCopy);
+            System.out.println(localReport);
+            Bills = BillsCopy;
+            addToLog(" Given " + amount + " UAH to a user");
+        } else {
+            System.out.println("There are no enough Bills !");
+        }
+    }
+
+    /**
+     * @param srcArray array contains information about different ATM bills before operation
+     * @param newArray array contains information about different ATM bills after operation
+     * @return string the contains information about how much and what nominal where given
+     */
+    private static String makeGiveReport(Bill[] srcArray, Bill[] newArray) {
+        StringBuilder report = new StringBuilder("");
```

Comment:
(thumb up)
The only notice, you don't need to initialize with an empty string, `new StringBuilder()` is enough